+nosnoc/+discrete_time_problem/Cls.m

```diff
@@ -0,0 +1,1600 @@
+            
+            x_prev = obj.w.x(0,0,opts.n_s);
+            for ii=1:opts.N_stages
+                h0 = obj.p.T().val/(opts.N_stages*opts.N_finite_elements(ii));
+                
+                ui = obj.w.u(ii);
+                p_stage = obj.p.p_time_var(ii);
+                p = [p_global;p_stage];
+                if obj.opts.use_speed_of_time_variables && opts.local_speed_of_time_variable
+                    s_sot = obj.w.sot(ii);
+                elseif obj.opts.use_speed_of_time_variables
+                    s_sot = obj.w.sot();
+                else
+                    s_sot = 1;
+                end
+                if opts.time_optimal_problem && ~opts.use_speed_of_time_variables
+                    t_stage = obj.w.T_final()/(opts.N_stages*opts.N_finite_elements(ii));
+                elseif opts.time_optimal_problem
+                    t_stage = s_sot*obj.p.T()/opts.N_stages;
+                else
+                    t_stage = obj.p.T()/opts.N_stages;
+                end
+
+                sum_h = 0;
+                for jj=1:opts.N_finite_elements(ii)
+                    if obj.opts.use_fesd
+                        h = obj.w.h(ii,jj);
+                        sum_h = sum_h + h;
+                    elseif opts.time_optimal_problem && ~opts.use_speed_of_time_variables
+                        h = obj.w.T_final()/(opts.N_stages*opts.N_finite_elements(ii));
+                    else
+                        h = h0;
+                    end
+
+                    % left boundary point handling
+                    q_prev = x_prev(1:dims.n_q);
+                    v_prev = x_prev(dims.n_q + 1:end);
+                    x_lbp = obj.w.x(ii,jj,0);
+                    q_lbp = x_lbp(1:dims.n_q);
+                    v_lbp = x_lbp(dims.n_q + 1:end);
+                    obj.g.q_continuity(ii,jj) = {q_prev - q_lbp};
+                    z_impulse_ij = obj.z_impulse(ii,jj);
+
+                    if (jj ~= 1 || ~opts.no_initial_impacts)
+                        obj.g.impulse(ii,jj) = {dcs.g_impulse_fun(q_lbp,v_lbp,v_prev,z_impulse_ij, v_global, p)};
+                    else
+                        obj.g.v_continuity(ii,jj) = {v_lbp-v_prev};
+                    end
+                    % additional y_eps constraint
+                    if opts.eps_cls > 0
+                        x_eps = vertcat(q_lbp + h * opts.eps_cls * v_lbp, v_lbp);
+                        obj.g.f_c_eps(ii,jj) = {dcs.f_c_fun(x_eps), 0, inf};
+                    end
+                    
+                    switch opts.rk_representation
+                      case RKRepresentation.integral
+                        % In integral representation stage variables are states.
+                        x_ij_end = opts.D_rk(1)*x_lbp;
+                        for kk=1:opts.n_s
+                            x_ijk = obj.w.x(ii,jj,kk);
+                            z_ijk = obj.w.z(ii,jj,kk);
```

Comment:
just to double check, these are user algebarics (and we have not CLS examples where they appear?)?

Reply:
Correct. Generally user algebraics in CLS are probably a very easy way to break ones model but gives us some flexibility.